### DISCUSSION TITLE
Edge Functions: Allow package.json without import_map.json

### DISCUSSION THREAD

**Discussion by jacob-israel-turner** [supabase/supabase#23239] [Feature Requests]
Hi friends!

I'm just getting started using Supabase edge functions. Most of my codebase is written for either a Node.js or browser environment, and I heavily utilize pnpm and package.json/pnpm for module resolution. I'm using an NX monorepo, so dependencies are pinned for all libraries and applications.

I was hoping to utilize Deno's [package.json](https://docs.deno.com/runtime/manual/node/package_json) compatibility out-of-the-box. In my editor, it seems to sync up to the correct version in my package.json automatically. However, once I run `supabase functions serve`, I get this error:

```
Relative import path "@oakserver/oak" not prefixed with / or ./ or ../ and not in import map from "file:///home/deno/functions/function-name/index.ts"
```

I assume this is because edge functions use their own runtime. Unfortunately I don't have much insight into the problem or solution, since this is my first time using Deno.

Fortunately, if I add this import map the problem goes away:
``` import_map.json
{
  "imports": {
    "@oakserver/oak": "npm:@oakserver/oak"
  }
}
```

One minor concern is that I'll need to do this for every node module that this function will rely on, which I expect to be significant. But, not a huge deal, I'll just have to fill it out.

The more pressing concern is that I'm not actually sure what _version_ this is pulling. Is it checking my package.json for the proper version, or just pulling the most recent version from NPM? If it's the latter, I suppose I'll have to pin the version in both my `package.json` _and_ `import_map.json` - which is going to be tedious and oft-forgotten.

If there's a way to make the edge functions environment properly pull the version from my package.json, please let me know! Thanks!

**Comment by dannybit** [comment]
What did you end up doing? I am facing a similar problem.

**Reply by carcinocron** [reply]
I stopped writing new code in supabase functions. The supabase backend client seems like it'll work in any nodejs serverless environment  (I'm using hono.dev and deploying to cloudflare workers). I have a pnpm monorepo and everything (except the supabase functions folder) is able to share types and functions.

**Reply by jacob-israel-turner** [reply]
I ended up abandoning supabase functions. I went with the simple option - deploying a docker container to fly.io.
